internal/json/types.go

```diff
@@ -311,16 +311,20 @@ func appendFloat(dst []byte, val float64, bitSize int) []byte {
 	case math.IsInf(val, -1):
 		return append(dst, `"-Inf"`...)
 	}
+	precision := FloatingPointPrecisionFunc()
```

Comment:
Would make more sense to pass the precision as an argument.

Reply:
So I was not able to set this as a global variable because that would introduce an import loop.

If you are OK with that, I can introduce a parameter to `appendFloat` and then pass it everywhere it is called.